elastic/kibana
# [App Search] Add multi-step validation to the Crawler Add Domain Form

## Summary

This migrates the multi-step URL validation for new domains from the standalone UX into Kibana. It could be helpful to [walk through each commit](https://github.com/elastic/kibana/pull/107251/commits).


## Screenshots

<img width="1000" alt="Screen Shot 2021-07-30 at 1 11 46 PM" src="https://user-images.githubusercontent.com/2479295/127688670-19a9e9e7-6160-4b4b-bcb4-38915ffb732e.png">

<img width="1000" alt="Screen Shot 2021-07-30 at 1 12 34 PM" src="https://user-images.githubusercontent.com/2479295/127688675-1922615b-62a8-4c1f-8033-3e083595550e.png">


## QA
1. Create a new engine or utilize an existing one
2. Go to the Crawler Overview
3. Click "Add domain"
- [ ] A flyout should appear
- [ ] The flyout's submit bttono should be disabled
- [ ] The validation steps are not visible
4.  Test some domains.  
- [ ] `elastic.co` should fail on the third and fourth step with blocking failures, so you can't submit.
- [ ]  `nymag.com` should have a single failure on the third step but its not blocking, you should be able to submit
5. After validation is completed and passes, you can submit your domain
- [ ] The flyout closes
- [ ] You're brought to the path for the Crawler single domain view for that domain

### Checklist
